debug/filt_02_unpack_fft.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Collect the output of a post-unpack snapshot and do an FFT on it.
+
+@author: paulp
+"""
+import argparse
+import sys
+import signal
+
+from casperfpga import katcp_fpga
+import matplotlib.pyplot as pyplot
+from casperfpga.memory import bin2fp
+import numpy
+
+FFT_CHANS = 4096
+
+parser = argparse.ArgumentParser(description='Perform an FFT on the incoming data.',
+                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('--host', dest='host', type=str, action='store', default='',
+                    help='the host to query')
+parser.add_argument('--integrate', dest='integrate', action='store', default=-1, type=int,
+                    help='integrate n successive spectra, -1 is infinite')
+parser.add_argument('--number', dest='number', action='store', default=-1, type=int,
+                    help='number of spectra/integrations to fetch, -1 is unlimited')
+parser.add_argument('--linear', dest='linear', action='store_true', default=False,
+                    help='plot linear, not log')
+parser.add_argument('--filtered', dest='filtered', action='store_true', default=False,
+                    help='plot filtered data, not raw ADC data')
+parser.add_argument('--loglevel', dest='log_level', action='store', default='',
+                    help='log level to use, default None, options INFO, DEBUG, ERROR')
+parser.add_argument('--checktvg', dest='checktvg', action='store_true', default='',
+                    help='if the TVG is enabled on the digitiser, check TVG data instead of generating'
+                         'an FFT on the data')
+args = parser.parse_args()
+
+import logging
+if args.log_level != '':
+    log_level = args.log_level.strip()
+    try:
+        logging.basicConfig(level=eval('logging.%s' % log_level))
+    except AttributeError:
+        raise RuntimeError('No such log level: %s' % log_level)
+LOGGER = logging.getLogger(__name__)
+
+def exit_gracefully(signal, frame):
+    fpga.disconnect()
+    sys.exit(0)
+signal.signal(signal.SIGINT, exit_gracefully)
+
+# make the FPGA objects
+fpga = katcp_fpga.KatcpFpga(args.host)
+fpga.get_system_information()
+
+# check for the required snapshot
+if 'snapadc0_ss' not in fpga.snapshots.names():
+    raise RuntimeError('ADC0 snap not available!')
+if 'snapadc1_ss' not in fpga.snapshots.names():
+    raise RuntimeError('ADC1 snap not available!')
+
+def get_data():
+    fpga.snapshots.snapadc0_ss.arm()
+    fpga.snapshots.snapadc1_ss.arm()
+    fpga.registers.ctrl2.write(trig_adc_snaps='pulse')
+    snapdata_p0 = fpga.snapshots.snapadc0_ss.read(arm=False)['data']
+    snapdata_p1 = fpga.snapshots.snapadc1_ss.read(arm=False)['data']
+    snapdata_p1['p1'] = []
+    for ctr, msb_data in enumerate(snapdata_p0['p1_msb']):
+        snapdata_p1['p1'].append((msb_data << 32) + snapdata_p1['p1_lsb'][ctr])
+
+    def eighty_to_ten(snapdata):
```

Comment:
Perhaps eighty_to_ten() belongs in a shared module? Seems to be quite popular :)